src/std/net/httpd/handler.ss

```diff
@@ -598,47 +642,74 @@ END-C
         ("TRACE"   'TRACE)
         ("OPTIONS" 'OPTIONS)))
 
-(def +http-response-codes+
-  (hash-eq (100 "Continue")
-           (101 "Switching Protocols")
-           (200 "OK")
-           (201 "Created")
-           (202 "Accepted")
-           (203 "Non-Authoritative Information")
-           (204 "No Content")
-           (205 "Reset Content")
-           (206 "Partial Content")
-           (300 "Multiple Choices")
-           (301 "Moved Permanently")
-           (302 "Found")
-           (303 "See Other")
-           (304 "Not Modified")
-           (305 "Use Proxy")
-           (307 "Temporary Redirect")
-           (400 "Bad Request")
-           (401 "Unauthorized")
-           (402 "Payment Required")
-           (403 "Forbidden")
-           (404 "Not Found")
-           (405 "Method Not Allowed")
-           (406 "Not Acceptable")
-           (407 "Proxy Authentication Required")
-           (408 "Request Timeout")
-           (409 "Conflict")
-           (410 "Gone")
-           (411 "Length Required")
-           (412 "Precondition Failed")
-           (413 "Request Entity Too Large")
-           (414 "Request-URI Too Long")
-           (415 "Unsupported Media Type")
-           (416 "Requested Range Not Satisfiable")
-           (417 "Expectation Failed")
-           (500 "Internal Server Error")
-           (501 "Not Implemented")
-           (502 "Bad Gateway")
-           (503 "Service Unavailable")
-           (504 "Gateway Timeout")
-           (505 "HTTP Version Not Supported")))
+(def +http-response-codes+ (make-hash-table-eq))
+(defclass http-condition ())
```

Comment:
make an interface for it so that we don't have to do method lookups.

Reply:
I was not sure how exactly to do it with Interfaces, so instead I made a struct with two fields for each condition.

Reply:
that's fine; but you got to learn how to utilize interface :)

Reply:
Oh I want to learn how to use them, for sure.